src/Microsoft.AspNetCore.Mvc.Core/Internal/PrefixContainer.cs

```diff
@@ -259,25 +247,62 @@ private static int IndexOfDelimiter(string entry, int startIndex)
             }
         }
 
-        private sealed class PrefixComparer : IComparer<string>
+        private int BinarySearch(string prefix)
         {
-            private readonly string _prefix;
-
-            public PrefixComparer(string prefix)
-            {
-                _prefix = prefix;
-            }
+            var start = 0;
+            var end = _sortedValues.Length - 1;
 
-            public int Compare(string x, string y)
+            while (start <= end)
             {
-                var testString = object.ReferenceEquals(x, _prefix) ? y : x;
-                if (IsPrefixMatch(_prefix, testString))
+                var pivot = start + (end - start >> 1);
+                var candidate = _sortedValues[pivot];
+                var compare = string.Compare(
+                    prefix,
+                    0,
+                    candidate,
+                    0,
+                    prefix.Length,
+                    StringComparison.OrdinalIgnoreCase);
+                if (compare == 0)
                 {
-                    return 0;
+                    // Ok, so now we have a candiate that for which candidate.StartsWith(prefix) is
```

Comment:
If we stick with string.compare above we should change this to reference the function it actually uses.